### PR TITLE
Feature branch acceptance test feedback

### PR DESCRIPTION
**Pull request by jgeurten**
Addresses the following issues/enhancements -

1. #130
2. #121
3. #117

### CODE REVIEW

File: WDAC-Policy-Wizard/app/MSIX/AllowMicrosoft - SingleFormat.xml [hunk 1]
```diff
@@ -0,0 +1,171 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (bowerscd-corp) [inline WDAC-Policy-Wizard/app/MSIX/AllowMicrosoft - SingleFormat.xml:1]: Why are we keeping single-format policies around? All in-support versions of windows support multi-policy?

File: WDAC-Policy-Wizard/app/MSIX/DefaultWindows_Audit - SingleFormat.xml [hunk 2]
```diff
@@ -0,0 +1,258 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<SiPolicy xmlns="urn:schemas-microsoft-com:sipolicy" PolicyType="Base Policy">
```
Comment (bowerscd-corp) [inline WDAC-Policy-Wizard/app/MSIX/DefaultWindows_Audit - SingleFormat.xml:2]: should not be here if this is truly single-policy format

File: WDAC-Policy-Wizard/app/src/ConfigTemplate_Control.cs [hunk 3]
```diff
@@ -505,23 +505,49 @@ private bool ReadSetRules(object sender, EventArgs e)
...
-                        break;
-
-                    case WDAC_Policy.NewPolicyTemplate.AllowMicrosoft:
-                        // Allow Microsoft mode
-                        xmlPathToRead = System.IO.Path.Combine(this._MainWindow.ExeFolderPath, "AllowMicrosoft.xml");
-                        break;
+                    // Multi-policy Format Policy Templates
+                    switch (this.Policy._PolicyTemplate)
+                    {
+                        case WDAC_Policy.NewPolicyTemplate.WindowsWorks:
```
Comment (bowerscd-corp) [inline WDAC-Policy-Wizard/app/src/ConfigTemplate_Control.cs:513]: formatting nitpick, `case` should be at same indentation level as brace

File: WDAC-Policy-Wizard/app/src/MainForm.cs [hunk 4]
```diff
@@ -1326,7 +1326,16 @@ public void MergeCustomRulesPolicy(List<string> customRulesPathList, string outp
...
                 {
-                    mergeScript += String.Format("\"{0}\",", path);
+                    // If the xml was successfully generated in the previous step
+                    if (File.Exists(path))
+                    {
+                        mergeScript += String.Format("\"{0}\",", path);
+                    }
+                    else
+                    {
+                        //this.nCustomValueRules--;
```
Comment (bowerscd-corp) [inline WDAC-Policy-Wizard/app/src/MainForm.cs:1336]: ?

File: WDAC-Policy-Wizard/app/src/MainForm.cs [hunk 5]
```diff
@@ -1510,29 +1519,26 @@ public bool ConvertPolicyToBinary()
             Pipeline pipeline = runspace.CreatePipeline();
 
             // If multi-policy format, use the {PolicyGUID}.cip format as defined in https://docs.microsoft.com/windows/security/threat-protection/windows-defender-application-control/deploy-multiple-windows-defender-application-control-policies#deploying-multiple-policies-locally
-            string binaryFileName; 
-            if(this.Policy._Format == WDAC_Policy.Format.MultiPolicy)
+            string binaryFileName = string.Empty;
+            SiPolicy finalSiPolicy = Helper.DeserializeXMLtoPolicy(this.Policy.SchemaPath);
```
Comment (bowerscd-corp) [inline WDAC-Policy-Wizard/app/src/MainForm.cs:1523]: Serialize Technically, Deserializing would be from bytes to xml

File: WDAC-Policy-Wizard/app/src/ConfigTemplate_Control.cs [hunk 6]
```diff
@@ -505,23 +505,49 @@ private bool ReadSetRules(object sender, EventArgs e)
...
+                            break;
+
+                        case WDAC_Policy.NewPolicyTemplate.SignedReputable:
+                            // Signed and Reputable Mode
+                            xmlPathToRead = System.IO.Path.Combine(this._MainWindow.ExeFolderPath, "SignedReputable - SingleFormat.xml");
+                            break;
+
+                        case WDAC_Policy.NewPolicyTemplate.AllowMicrosoft:
+                            // Allow Microsoft mode
+                            xmlPathToRead = System.IO.Path.Combine(this._MainWindow.ExeFolderPath, "AllowMicrosoft - SingleFormat.xml");
```
Comment (bowerscd-corp) [inline WDAC-Policy-Wizard/app/src/ConfigTemplate_Control.cs:546]: It seems like there should be a better way than to hard-code these values as strings, here.

File: WDAC-Policy-Wizard/app/MSIX/AllowMicrosoft - SingleFormat.xml (hunk 1 above)
Comment (jsuther1974) [reply WDAC-Policy-Wizard/app/MSIX/AllowMicrosoft - SingleFormat.xml:1]: Not true. Server 2016 and 2019 are in service

**Review by bowerscd-corp — COMMENTED** [review]
I would recommend cleaning up your git commits so that `git bisect` can be used to find bugs.

File: WDAC-Policy-Wizard/app/MSIX/AllowMicrosoft - SingleFormat.xml (hunk 1 above)
Comment (bowerscd-corp) [reply WDAC-Policy-Wizard/app/MSIX/AllowMicrosoft - SingleFormat.xml:1]: WS2019 doesn't support Multi-policy?

Comment (jsuther1974) [reply WDAC-Policy-Wizard/app/MSIX/AllowMicrosoft - SingleFormat.xml:1]: Sadly no. It's based on 1809. Multi policy was introduced in 1903

File: WDAC-Policy-Wizard/app/src/MainForm.cs (hunk 4 above)
Comment (jgeurten) [reply WDAC-Policy-Wizard/app/src/MainForm.cs]: Fixed

File: WDAC-Policy-Wizard/app/src/ConfigTemplate_Control.cs (hunk 3 above)
Comment (jgeurten) [reply WDAC-Policy-Wizard/app/src/ConfigTemplate_Control.cs]: Woah, never knew that. Thanks!

File: WDAC-Policy-Wizard/app/src/MainForm.cs (hunk 5 above)
Comment (jgeurten) [reply WDAC-Policy-Wizard/app/src/MainForm.cs:1515]: Is this true? Summary for System.Xml.Serialization implies the opposite:

Summary:
        //     Deserializes the XML document contained by the specified System.IO.TextReader.
        //
        // Parameters:
        //   textReader:
        //     The System.IO.TextReader that contains the XML document to deserialize.
        //
        // Returns:
        //     The System.Object being deserialized.

Comment (bowerscd-corp) [reply WDAC-Policy-Wizard/app/src/MainForm.cs:1515]: https://en.wikipedia.org/wiki/Serialization

> The opposite operation, **extracting a data structure from a series of bytes**, is deserialization, (also called unserialization or [unmarshalling](https://en.wikipedia.org/wiki/Unmarshalling)).

File: WDAC-Policy-Wizard/app/src/ConfigTemplate_Control.cs (hunk 6 above)
Comment (bowerscd-corp) [reply WDAC-Policy-Wizard/app/src/ConfigTemplate_Control.cs]: You can put them in the resource.resx, or as a static internal const string in a namespace somewhere to keep them private to the assembly.

Comment (bowerscd-corp) [reply WDAC-Policy-Wizard/app/src/ConfigTemplate_Control.cs]: if the goal is to make these starter templates, I'd recommend probably just compiling the XML into the binary itself as a resource.

Comment (jgeurten) [reply WDAC-Policy-Wizard/app/src/ConfigTemplate_Control.cs]: yeah, this is a better approach. I will just package the xml as a resource

**Pull request merged**